tsdb/chunkenc/chunk.go

```diff
@@ -49,6 +49,7 @@ type Chunk interface {
 	// be re-used or a new iterator can be allocated.
 	Iterator(Iterator) Iterator
 	NumSamples() int
+	Compact()
```

Comment:
I believe we need some comment on what we expect from this method for someone who will add some new implementation soon.

Reply:
If you will be kind enough to add some simple comments and all of others comments for GoDoc.. it would be amazing :heart: 

[Scout rule](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)!

Reply:
Hm I am also thinking if we can separate the chunk interface to reader and writer ): 

It's too wide to me..

Reply:
> If you will be kind enough to add some simple comments and all of others comments for GoDoc

Sure, done. May you take a look? Feedback is welcome 🙏

Reply:
> Hm I am also thinking if we can separate the chunk interface to reader and writer ):
> It's too wide to me..

Few questions, please:
1. Do you think it's good to couple such refactoring within this optimization PR?
2. If so, may you share more details about your thoughts? Are you thinking about keeping the current `Chunk` and add an "read-only" interface?

Reply:
No, I think I am ok to merge this and think about it later :+1: 

> If so, may you share more details about your thoughts? Are you thinking about keeping the current Chunk and add a "read-only" interface?

No..  I think about two separate interfaces as this interface is becoming too wide. Usually, things that depend on this interface either only read or only write.